test/schemas.test.js

```diff
@@ -269,3 +269,45 @@ test('Triple $ref deep', t => {
     t.deepEquals(JSON.parse(res.payload).message, "body should have required property 'foo'")
   })
 })
+
+test('$ref with a simple $id', t => {
+  t.plan(4)
+  const fastify = Fastify()
+  const ajv = new AJV()
+  ajv.addSchema(fastClone(schemaUsed))
+  ajv.addSchema({
+    $id: 'urn:schema:response',
+    type: 'object',
+    required: ['foo'],
+    properties: {
+      foo: { $ref: 'urn:schema:foo' }
+    }
+  })
+
+  fastify.setSchemaCompiler(schema => ajv.compile(schema))
+  fastify.setSchemaResolver((ref) => {
+    t.equals(ref, 'urn:schema:foo')
+    return ajv.getSchema(ref).schema
+  })
+
+  fastify.route({
+    method: 'POST',
+    url: '/',
+    schema: {
+      body: ajv.getSchema('urn:schema:response').schema
```

Comment:
Actually, can you make this a response schema? This test won’t hit `fast-json-stringify` otherwise. And that needs to happen to cover the full scenario.

Reply:
I found the problem: the `$ref = #/definitions/foo`  in `urn:schema:foo` is consider relative to `urn:schema:response`, and this is wrong.

Looking at how to solve